zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/client/TenantClient.java

```diff
@@ -279,6 +279,11 @@ public TenantAddEntityClient withEntityType(final EntityType entityType) {
       return this;
     }
 
+    public TenantAddEntityClient withTenantId(final String tenantId) {
+      tenantRecord.setTenantId(tenantId);
+      return this;
+    }
```

Comment:
❌ We shouldn't need this. When assigning an entity to a tenant it works with keys so tenant id shouldn't need to be set.

Reply:
What do you mean, we add the `tenantId` to the user

Reply:
What I would expect is that we use the key to get the tenant from the state. In the state we already have tenant id store. So basically, get the `tenantId` from the `persistedRecord`, instead of the `record` 🙂

Reply:
Looking at the API spec we won't be getting a tenant id in the command:

```yaml
  /tenants/{tenantKey}/users/{userKey}:
    put:
      tags:
        - Tenant
      operationId: assignUserToTenant
      summary: Assign a user to a tenant
      description: Assign a single user to a specified tenant.
      parameters:
        - name: tenantKey
          in: path
          required: true
          description: The unique identifier of the tenant.
          schema:
            type: integer
            format: int64
        - name: userKey
          in: path
          required: true
          description: The unique identifier of the user.
          schema:
            type: integer
            format: int64
```

Reply:
I'll fix the bug in the processor and remove the test helper method 👍